src/components/structures/MessagePanel.js

```diff
@@ -368,6 +369,15 @@ module.exports = React.createClass({
         this.eventNodes[eventId] = node;
     },
 
+    // once dynamic content in the events load, make the scrollPanel check the
+    // scroll offsets.
+    _onWidgetLoad: function() {
+        var scrollPanel = this.refs.scrollPanel;
+        if (scrollPanel) {
+            scrollPanel.checkScroll();
```

Comment:
I don't *think* this is sufficient - `GeminiPanel` needs a `forceUpdate` so that it can resize the scrollbar.